### PR TITLE
UI: Add feature to download SBOM scan results

### PR DESCRIPTION
**Pull request by davakos**
## Description
Adds an option to download SBOM results from the Scan Results Raw tab if the scan included SBOM scan data.
This download will only include the SBOM JSON data, it will not include the additional scan information, such as repo, branch, scan_options, etc.

Other updates in this PR:
1. Updates validation schemas to more closely align to underlying TypeScript types/interfaces and concat schemas to reduce re-declaring validation fields
2. Updates third party dependencies
   - update `actions/checkout` to 3.5.0. Note: 2.x=>3.x update was to use NodeJS 16 runtime by default which requires a `minimum Actions Runner version of v2.285.0 to run, which is by default available in GHES 3.4 or later`
   - add `@babel/types` as a `devDependency` to bump version to suppress deprecation warning: `Trace: isModuleDeclaration has been deprecated, please migrate to isImportOrExportDeclaration`
3. Fix features validation bug on Users page 
4. Bumps version to 1.13

## Motivation and Context
Allows saving SBOM results via UI

## How Has This Been Tested?
1. Tested SBOM download button appears when scan contains SBOM data
2. Tested SBOM results download
3. Tested updated validation schemas

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [X] My code follows conforms to the coding standards.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes. (Pending)
- [X] All new and existing tests passed.

## Pic
Embed something funny here

### CODE REVIEW

**Review by g-marconet — APPROVED** [review]

File: ui/src/components/TableMenu.test.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ describe("TableMenu component", () => {
 			"1"
 		);
 
-		return expect(spy).toHaveBeenCalledWith(exportFile, data, undefined);
```
Comment (davakos) [inline ui/src/components/TableMenu.test.tsx:102]: remove premature returns, oopsie!

File: ui/src/features/scans/scansSchemas.ts [hunk 2]
```diff
@@ -381,7 +381,6 @@ const scanResultsSchema: Yup.ObjectSchema<ScanResults> = Yup.object()
 	.defined();
 
 export const scanIdSchema = Yup.string()
-	.defined()
```
Comment (davakos) [inline ui/src/features/scans/scansSchemas.ts:339]: schema used for validating scan ids & batch ids. removed `defined()` in this common schema so batch ids can be missing in the scan data. where scan id fields are validated, they already include their own `defined()` check

File: ui/src/pages/ResultsPage.tsx [hunk 3]
```diff
@@ -4623,7 +4622,7 @@ interface AllStylesT {
 
 type DownloadType = "sbom" | "scan";
 
-const CodeTabContent = (props: {
+export const CodeTabContent = (props: {
```
Comment (davakos) [inline ui/src/pages/ResultsPage.tsx:4625]: for testing

File: ui/src/pages/ResultsPage.tsx [hunk 4]
```diff
@@ -4684,7 +4683,7 @@ const CodeTabContent = (props: {
 		}
 
 		try {
-			exportToJson(type, data);
+			formatters.exportToJson(type, data);
```
Comment (davakos) [inline ui/src/pages/ResultsPage.tsx:4686]: called within `formatters` instead of directly so we can mock & spy on it in unit tests

**Review by g-marconet — APPROVED** [review]

**Pull request merged**